### ISSUE TITLE
Chat history

### ISSUE THREAD

**Issue by Tamriel** [qTox/qTox#281]

**Comment by apprb** [comment]
Probably this is a good place to discuss about logs. 
For non-encrypted solution everything is simple: just use sqlite (already tried it in my history branch). 
Problem in encrypted logs. I found 2 possible solutions: 
1. https://github.com/devbean/QtCipherSqlitePlugin Looks good but not easy to install.
2. https://www.zetetic.net/sqlcipher/ (I prefer this variant) No problems for *nix systems (presents in debian, ubuntu repos) but I don't know what about other systems.

Why sqlite: fast, easy to parse logs, easy to clean, portable (at least non-encrypted variant).

My additional question: is it possible to use toxcore infrastructure to encrypt/decrypt files?

**Comment by dubslow** [comment]
I can take care of encryption by refactoring encrypted_save to encrypt arbitrary data instead of just save files. It would be password protected; using whatever-key instead of a password is already possible (see crypto_core.h).

tl;dr all encryption should be done through toxcore, no additional dependencies.

**Comment by apprb** [comment]
OK, then in the following few days I'll prepare something easy to encryption add.

**Comment by dubslow** [comment]
@apprb https://github.com/dubslow/toxcore/commit/e6f30694d32a81f9171b2057d9c873cc16f6dca1#diff-2

**Comment by apprb** [comment]
I'll take it into account. Thank you.

**Comment by dubslow** [comment]
An expanded API has been merged 
https://github.com/irungentoo/toxcore/blob/master/toxencryptsave/toxencryptsave.h

**Comment by agilob** [comment]
close
